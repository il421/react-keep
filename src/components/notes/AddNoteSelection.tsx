import React from "react";

export class AddNoteSelection extends React.PureComponent {
  render() {
    return (
      <>
        AddNoteSelection
      </>
    );
  }
}
