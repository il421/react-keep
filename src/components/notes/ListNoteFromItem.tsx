import React from "react";
import { FlexBox } from "../ui-components/FlexBox";
import { AlignItems, JustifyContent } from "../../common/variables";
import { CheckboxInputField } from "../form/CheckboxInputField";
import { TextInputField } from "../form/TextInputField";
import { IconButton } from "../ui-components/IconButton";
import "../../styles/components/notes/_list-note-form-item.scss";

interface ListNoteFromItemProps {
  name: string;
  onRemove: () => void;
}

export const ListNoteFromItem: React.FunctionComponent<ListNoteFromItemProps> = ({
  name,
  onRemove
}) => {
  return (
    <FlexBox
      justifyContent={JustifyContent.spaceBetween}
      alignItems={AlignItems.start}
      className="list-note-form-item"
    >
      {/* @TODO for drag and drop */}
      {/*<FontAwesomeIcon*/}
      {/*  icon={"th"}*/}
      {/*  className="list-note-form-item__mover"*/}
      {/*  size="lg"*/}
      {/*/>*/}
      <CheckboxInputField
        name={`${name}.checked`}
        className="list-note-form-item__check"
        id={`${name}.checked`}
      />
      <TextInputField
        isTextArea={true}
        name={`${name}.content`}
        placeholder="Type something ..."
        className="list-note-form-item__text"
      />
      <IconButton
        onButtonClick={onRemove}
        icon={"times"}
        className="list-note-form-item__remove"
      />
    </FlexBox>
  );
};
