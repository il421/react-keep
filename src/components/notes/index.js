export * from "./NotesList";
export * from "./NotesItem";
export * from "./notes.types";
export * from "./TextNoteFormModal";
export * from "./ListNoteFormModal";
export * from "./NoteForm";
export * from "./options/ColorsPickerField";
