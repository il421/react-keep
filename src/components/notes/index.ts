export * from "./ConfirmDialog";
export * from "./ImageNoteFormModal";
export * from "./ListNoteFormModal";
export * from "./ListNoteFromItem";
export * from "./NoteContent";
export * from "./NoteForm";
export * from "./notes.types";
export * from "./NotesItem";
export * from "./TextNoteFormModal";
export * from "./utils";
export * from "./options/ColorsPickerField";
export * from "./options/TagsPickerField";
