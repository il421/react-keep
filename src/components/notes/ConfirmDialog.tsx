import React, { Dispatch } from "react";
import { connect } from "react-redux";
import { handleRemoveNote } from "../../actions/notes";

/**
 * This component works only as a part of Note for now.
 */

interface ConfirmDialogProps extends Pick<HTMLElement, "className">{
  id: string;
  removeNote: (id: string) => void;
  closeDialog: () => void;
}

const ConfirmDialog: React.FunctionComponent<ConfirmDialogProps> = ({ className, closeDialog, removeNote, id }) => {
  return (
    <div className={className}>
      <button type="button"
        onClick={() => {
          removeNote(id);
          closeDialog();
        }
        }
      >Yes</button>
      <button type="button" onClick={closeDialog}>No</button>
    </div>
  );
};


const mapDispatchToProps = (dispatch: Dispatch<any>) => ({
  removeNote: (id: string) => dispatch(handleRemoveNote(id))
});


export default connect(undefined, mapDispatchToProps)(ConfirmDialog);
