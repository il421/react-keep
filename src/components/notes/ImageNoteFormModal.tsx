import React from "react";
import Modal from "react-modal";
import { NoteFormValues, NoteType } from "./notes.types";
import {
  isModal,
  nameOf,
  Placeholders,
  AlignItems,
  JustifyContent,
  Errors,
} from "../../common";
import { PathNames, QueryKeys } from "../../routers/Routing";
import { FileFormField, TextInputField } from "../form";
import "../../styles/components/notes/_note-form.scss";
import "../../styles/components/notes/_image-note-form-item.scss";
import { NoteForm } from "./NoteForm";
import { History } from "history";
import { ImageItem } from "../../store/store.types";
import { Field, FormRenderProps, FormSpy } from "react-final-form";
import { IconButton, FlexBox } from "../ui-components";
import { ValidationErrors } from "final-form";

export interface ImageNoteFormModalProps {
  history: History;
}

export const getValidationErrors = (
  values: Pick<NoteFormValues<ImageItem>, "content">
): ValidationErrors => {
  const errors: ValidationErrors = { content: {} };
  const { uploadedImage, imageUrl } = values.content!;

  if (!uploadedImage && !imageUrl) {
    errors.content.uploadedImage = Errors.required;
  }

  return errors;
};

export class ImageNoteFormModal extends React.Component<
  ImageNoteFormModalProps
> {
  private nameOf = nameOf<NoteFormValues<ImageItem>>();
  private nameOfContent = nameOf<ImageItem>();
  private readonly NO_IMAGE_URL: string = "img/no_image.png";
  private readonly FIELD_ID: string = "image";

  private handleUploadClick = () => {
    const fileInput = document.getElementById(this.FIELD_ID);
    fileInput && fileInput.click();
  };

  render() {
    if (
      !isModal({
        query: this.props.history.location.search,
        type: QueryKeys.image,
      })
    ) {
      return null;
    }

    return (
      <Modal
        isOpen={true}
        onRequestClose={() => this.props.history.push(PathNames.base)}
        closeTimeoutMS={200}
        className="note-modal"
        ariaHideApp={false}
      >
        <NoteForm
          type={NoteType.image}
          history={this.props.history}
          validate={getValidationErrors}
        >
          <FormSpy>
            {({
              form,
              values,
              initialValues,
            }: FormRenderProps<NoteFormValues<ImageItem>>) => {
              return (
                <div className="image-note-form-item">
                  <Field
                    name={`${this.nameOf("content")}.${this.nameOfContent(
                      "imageUrl"
                    )}`}
                  >
                    {() => (
                      <img
                        className="image-note-form-item__img"
                        src={
                          values.content.imageUrl &&
                          !values.content.uploadedImage
                            ? values.content.imageUrl
                            : values.content.uploadedImage
                            ? URL.createObjectURL(values.content.uploadedImage)
                            : this.NO_IMAGE_URL
                        }
                        alt="Uploaded file"
                      />
                    )}
                  </Field>
                  {!initialValues.createdBy && (
                    <FlexBox
                      className="image-note-form-item__btns"
                      justifyContent={JustifyContent.spaceBetween}
                      alignItems={AlignItems.center}
                      vertical
                    >
                      <IconButton
                        icon="upload"
                        onButtonClick={() => {
                          this.handleUploadClick();
                        }}
                      />
                      <IconButton
                        icon="times"
                        onButtonClick={() => {
                          form.change(
                            `${this.nameOf("content")}[${this.nameOfContent(
                              "uploadedImage"
                            )}]`,
                            undefined
                          );
                        }}
                      />
                      {initialValues.content.imageUrl && (
                        <IconButton
                          id="test-trash-image-button"
                          icon="trash"
                          onButtonClick={() => {
                            form.change(
                              `${this.nameOf("content")}[${this.nameOfContent(
                                "imageUrl"
                              )}]`,
                              null
                            );
                            form.change(
                              `${this.nameOf("content")}[${this.nameOfContent(
                                "uploadedImage"
                              )}]`,
                              undefined
                            );
                          }}
                        />
                      )}
                      <FileFormField
                        id={this.FIELD_ID}
                        name={`${this.nameOf("content")}.${this.nameOfContent(
                          "uploadedImage"
                        )}]`}
                        className="image-note-form-item__field"
                      />
                    </FlexBox>
                  )}
                </div>
              );
            }}
          </FormSpy>

          <TextInputField
            isTextArea={true}
            name={`${this.nameOf("content")}.${this.nameOfContent("text")}`}
            placeholder={Placeholders.content}
            className="note-form__text note-form__text--image"
          />
        </NoteForm>
      </Modal>
    );
  }
}
