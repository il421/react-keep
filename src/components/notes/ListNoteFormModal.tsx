import React from "react";
import Modal from "react-modal";
import { ListItem } from "../../store/store.types";
import { NoteFormValues, NoteType } from "./notes.types";
import { PathNames, QueryKeys, RouteActions } from "../../routers/Routing";
import { NoteForm } from "./NoteForm";
import { History } from "history";
import { FieldArray, FieldArrayRenderProps } from "react-final-form-arrays";

import { ListNoteFromItem } from "./ListNoteFromItem";
import { FormRenderProps, FormSpy } from "react-final-form";
import { FlexBox, IconButton } from "../ui-components";
import {
  AlignItems,
  isModal,
  JustifyContent,
  nameOf,
  newLineRegEx,
} from "../../common";
import { getDefaultContent } from "./utils";
import { parse } from "query-string";

export interface ListNoteFormModalProps {
  history: History;
}

export class ListNoteFormModal extends React.Component<ListNoteFormModalProps> {
  private nameOf = nameOf<NoteFormValues<ListItem[]>>();

  private getCheckedItems = (content: ListItem[]): string => {
    if (!content) return "";
    return `${content.length}/${content.filter((c) => c.checked).length}`;
  };

  private isNewNote: boolean =
    parse(this.props.history.location.search)[QueryKeys.list] ===
    RouteActions.add;

  render() {
    if (
      !isModal({
        query: this.props.history.location.search,
        type: QueryKeys.list,
      })
    ) {
      return null;
    }
    return (
      <Modal
        isOpen={true}
        onRequestClose={() => this.props.history.push(PathNames.base)}
        closeTimeoutMS={200}
        className="note-modal"
        ariaHideApp={false}
      >
        <NoteForm type={NoteType.list} history={this.props.history}>
          <FormSpy>
            {({
              form,
              values,
            }: FormRenderProps<NoteFormValues<ListItem[]>>) => {
              const addItem = () => {
                form.change(this.nameOf("content"), [
                  getDefaultContent<ListItem[]>(NoteType.list)[0],
                  ...values.content,
                ]);
              };
              return (
                <>
                  <FlexBox
                    justifyContent={JustifyContent.spaceBetween}
                    alignItems={AlignItems.center}
                  >
                    <div id="test-checked-items">
                      Checked: {this.getCheckedItems(values.content)}
                    </div>
                    <IconButton
                      onButtonClick={addItem}
                      icon="plus-circle"
                      text="Add line"
                      id="test-add-item-button"
                    />
                  </FlexBox>

                  <div className="note-form__list">
                    <FieldArray name={this.nameOf("content")}>
                      {(
                        props: FieldArrayRenderProps<ListItem, HTMLElement> & {
                          fields: {
                            sortArray: (
                              compareFn: (a: any, b: any) => number
                            ) => void;
                          };
                        }
                      ) => {
                        const handleOnChecked = () => {
                          props.fields.sortArray(
                            (a: ListItem, b: ListItem) =>
                              Number(a.checked) - Number(b.checked)
                          );
                        };

                        const handleOnRemove = (index: number) => () => {
                          props.fields.remove(index);
                        };

                        const setPastedValue = (index: number) => (
                          paste: string
                        ) => {
                          const values: string[] = paste
                            .split(newLineRegEx)
                            .filter((v) => !!v);

                          if (values.length > 1) {
                            const defaultItem: ListItem = getDefaultContent<
                              ListItem[]
                            >(NoteType.list)[0];
                            form.batch(() => {
                              for (let i = 0; i < values.length; i++) {
                                if (i === 0) {
                                  // update the first element
                                  props.fields.update(index, {
                                    ...defaultItem,
                                    content: values[i].trim(),
                                  });
                                } else {
                                  // push others
                                  props.fields.push({
                                    ...defaultItem,
                                    content: values[i].trim(),
                                  });
                                }
                              }
                            });
                          }
                        };

                        return props.fields.map(
                          (name: string, index: number) => (
                            <ListNoteFromItem
                              autoFocus={this.isNewNote && index === 0}
                              name={name}
                              key={props.fields.length! - index} // to focus the first element
                              index={index}
                              onRemove={handleOnRemove(index)}
                              addItem={addItem}
                              onChecked={handleOnChecked}
                              setPastedValue={setPastedValue(index)}
                            />
                          )
                        );
                      }}
                    </FieldArray>
                  </div>
                </>
              );
            }}
          </FormSpy>
        </NoteForm>
      </Modal>
    );
  }
}
