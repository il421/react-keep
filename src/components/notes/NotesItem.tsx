import React, { useState } from "react";
import moment from "moment";
import { Note } from "../../store/store.types";
import { ContentContainer, FlexBox, IconButton } from "../ui-components";
import { AlignItems, Colors, JustifyContent } from "../../common/variables";
import "../../styles/components/notes/_note.scss";
import ConfirmDialog from "./ConfirmDialog";

export interface NoteProps {
  note: Note
}

export const NotesItem:React.FunctionComponent<NoteProps> = ({ note }): JSX.Element => {

  const [isConfirm, setIsConfirm] = useState<boolean>(false);

  return (
    <ContentContainer className="note note--animation" style={{ backgroundColor: note.color }}>
      <FlexBox vertical justifyContent={JustifyContent.spaceBetween}>
        {/*actions*/}
        <FlexBox className="note__actions" justifyContent={JustifyContent.spaceBetween} alignItems={AlignItems.center}>
          <IconButton
            onClick={() => {}}
            icon="bookmark"
            color={ note.important ? Colors.varmillion : Colors.lightGray }
          />
          <IconButton onClick={() => setIsConfirm(true)} icon="times" />
        </FlexBox>

        {/*title and content*/}
        <FlexBox className="note__content content" vertical justifyContent={JustifyContent.spaceBetween}>
          { !!note.title.trim() && <h2 className="content__title">{ note.title }</h2> }
          <div className="content__text">{ note.content }</div>
        </FlexBox>

        {/*date nad tags*/}
        <FlexBox className="note__details details"  vertical justifyContent={JustifyContent.spaceBetween}>
          <div className="details__date">{ moment(note.updatedAt).format("MMMM Do, YYYY") }</div>

          <FlexBox className="details__tags" justifyContent={JustifyContent.start}>
            tags
          </FlexBox>
        </FlexBox>
      </FlexBox>

      {
        isConfirm && (
          <ConfirmDialog className="note__confirm" id={ note.id } closeDialog={ () => setIsConfirm(false) } />
        )
      }
    </ContentContainer>
  );
};
