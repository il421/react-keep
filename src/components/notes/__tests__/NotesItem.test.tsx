import React from "react";
import { ReactWrapper } from "enzyme";
import { NotesItem, NoteProps } from "../NotesItem";
import { notes } from "../../../testData/notes";
import { NoteType } from "../notes.types";
import { flushPromises, mountInApp } from "../../../common/testUtils";
import { collaborators } from "../../../testData/users";
import { Collaborator } from "../../../store/store.types";

let wrapper: ReactWrapper | undefined, props: NoteProps;

beforeEach(() => {
  props = {
    note: notes[3],
    removeNote: jest.fn(),
    moveToArchive: jest.fn(),
    toggleImportance: jest.fn(),
    onNoteSelected: jest.fn(),
    getCollaborator: jest.fn(),
  };

  wrapper = mountInApp(<NotesItem {...props} />);
});

afterEach(() => {
  if (wrapper) {
    wrapper.unmount();
    wrapper = undefined;
  }
});

describe("Rendering", () => {
  test("should render NotesItem correctly", () => {
    if (wrapper) {
      expect(wrapper.debug()).toMatchSnapshot();
    }
  });
});

describe("Actions", () => {
  test("should select note", () => {
    if (wrapper) {
      wrapper.find(".note").hostNodes().simulate("click");
      expect(props.onNoteSelected).toHaveBeenLastCalledWith(NoteType.text, "4");
    }
  });

  test("should toggle importance", () => {
    if (wrapper) {
      wrapper.find("#test-toggle-importance-4").hostNodes().simulate("click");
      expect(props.toggleImportance).toHaveBeenLastCalledWith("4");
    }
  });

  test("should move a note to archive", () => {
    if (wrapper) {
      wrapper.find("#test-toggle-arch-4").hostNodes().simulate("click");
      wrapper.update();

      wrapper
        .find("#test-confirm-dialog-delete-btn")
        .hostNodes()
        .simulate("click");

      expect(props.moveToArchive).toHaveBeenLastCalledWith("4");
    }
  });

  test("should delete a note to archive", () => {
    if (wrapper) {
      wrapper.find("#test-delete-note-4").hostNodes().simulate("click");
      wrapper.update();

      wrapper
        .find("#test-confirm-dialog-delete-btn")
        .hostNodes()
        .simulate("click");

      expect(props.removeNote).toHaveBeenLastCalledWith("4");
    }
  });

  test("should show collaborators icons", async () => {
    wrapper = mountInApp(
      <NotesItem
        {...props}
        note={notes[0]}
        getCollaborator={(uid) =>
          collaborators.find((c: Collaborator) => c.uid === uid)
        }
      />
    );
    wrapper.update();

    if (wrapper) {
      expect(wrapper.find(".coin").length).toBe(4);
    }
  });
});
