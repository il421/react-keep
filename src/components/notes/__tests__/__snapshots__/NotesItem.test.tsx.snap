// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering should render NotesItem correctly 1`] = `
"<WithFontAwesome>
  <Provider store={{...}}>
    <NotesItem note={{...}} removeNote={[Function: mockConstructor]} moveToArchive={[Function: mockConstructor]} toggleImportance={[Function: mockConstructor]} onNoteSelected={[Function: mockConstructor]} getCollaborator={[Function: mockConstructor]}>
      <ContentContainer className=\\"note\\" style={{...}} onClick={[Function: onClick]}>
        <div onClick={[Function: onClick]} className=\\"content-container note\\" style={{...}}>
          <FlexBox vertical={true} justifyContent=\\"space-between\\">
            <div style={{...}} className={[undefined]}>
              <FlexBox className=\\"note__actions actions\\" justifyContent=\\"space-between\\" alignItems=\\"center\\">
                <div style={{...}} className=\\"note__actions actions\\">
                  <IconButton className=\\"actions__important\\" onButtonClick={[Function: onButtonClick]} icon=\\"bookmark\\" color=\\"#7A7A7A\\" id=\\"test-toggle-importance-4\\">
                    <button id=\\"test-toggle-importance-4\\" className=\\"icon-button actions__important \\" onClick={[Function: onButtonClick]} disabled={[undefined]} type=\\"button\\">
                      <FontAwesomeIcon icon=\\"bookmark\\" color=\\"#7A7A7A\\" border={false} className=\\"\\" mask={{...}} fixedWidth={false} inverse={false} flip={{...}} listItem={false} pull={{...}} pulse={false} rotation={{...}} size={{...}} spin={false} symbol={false} title=\\"\\" transform={{...}} swapOpacity={false}>
                        <svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"bookmark\\" className=\\"svg-inline--fa fa-bookmark fa-w-12 \\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 384 512\\" style={{...}} color=\\"#7A7A7A\\">
                          <path fill=\\"currentColor\\" d=\\"M0 512V48C0 21.49 21.49 0 48 0h288c26.51 0 48 21.49 48 48v464L192 400 0 512z\\" style={{...}} />
                        </svg>
                      </FontAwesomeIcon>
                    </button>
                  </IconButton>
                  <FlexBox justifyContent=\\"flex-start\\" flexGrow={true} className=\\"actions__col-list\\">
                    <div style={{...}} className=\\"actions__col-list\\" />
                  </FlexBox>
                  <FlexBox justifyContent=\\"flex-start\\" alignItems=\\"center\\">
                    <div style={{...}} className={[undefined]}>
                      <IconButton className=\\"actions__archive\\" onButtonClick={[Function: onButtonClick]} icon=\\"archive\\" id=\\"test-toggle-arch-4\\">
                        <button id=\\"test-toggle-arch-4\\" className=\\"icon-button actions__archive \\" onClick={[Function: onButtonClick]} disabled={[undefined]} type=\\"button\\">
                          <FontAwesomeIcon icon=\\"archive\\" border={false} className=\\"\\" mask={{...}} fixedWidth={false} inverse={false} flip={{...}} listItem={false} pull={{...}} pulse={false} rotation={{...}} size={{...}} spin={false} symbol={false} title=\\"\\" transform={{...}} swapOpacity={false}>
                            <svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"archive\\" className=\\"svg-inline--fa fa-archive fa-w-16 \\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 512 512\\" style={{...}}>
                              <path fill=\\"currentColor\\" d=\\"M32 448c0 17.7 14.3 32 32 32h384c17.7 0 32-14.3 32-32V160H32v288zm160-212c0-6.6 5.4-12 12-12h104c6.6 0 12 5.4 12 12v8c0 6.6-5.4 12-12 12H204c-6.6 0-12-5.4-12-12v-8zM480 32H32C14.3 32 0 46.3 0 64v48c0 8.8 7.2 16 16 16h480c8.8 0 16-7.2 16-16V64c0-17.7-14.3-32-32-32z\\" style={{...}} />
                            </svg>
                          </FontAwesomeIcon>
                        </button>
                      </IconButton>
                      <IconButton className=\\"actions__remove\\" onButtonClick={[Function: onButtonClick]} icon=\\"times\\" id=\\"test-delete-note-4\\">
                        <button id=\\"test-delete-note-4\\" className=\\"icon-button actions__remove \\" onClick={[Function: onButtonClick]} disabled={[undefined]} type=\\"button\\">
                          <FontAwesomeIcon icon=\\"times\\" border={false} className=\\"\\" mask={{...}} fixedWidth={false} inverse={false} flip={{...}} listItem={false} pull={{...}} pulse={false} rotation={{...}} size={{...}} spin={false} symbol={false} title=\\"\\" transform={{...}} swapOpacity={false}>
                            <svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"times\\" className=\\"svg-inline--fa fa-times fa-w-11 \\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 352 512\\" style={{...}}>
                              <path fill=\\"currentColor\\" d=\\"M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z\\" style={{...}} />
                            </svg>
                          </FontAwesomeIcon>
                        </button>
                      </IconButton>
                    </div>
                  </FlexBox>
                </div>
              </FlexBox>
              <FlexBox className=\\"note__content content\\" vertical={true} justifyContent=\\"space-between\\">
                <div style={{...}} className=\\"note__content content\\">
                  <h2 className=\\"content__title\\">
                    4
                  </h2>
                  <NoteContent type={0} content=\\"text\\">
                    <div className=\\"content__text\\">
                      text
                    </div>
                  </NoteContent>
                </div>
              </FlexBox>
              <FlexBox className=\\"note__details details\\" vertical={true} justifyContent=\\"space-between\\">
                <div style={{...}} className=\\"note__details details\\">
                  <FlexBox className=\\"details__date\\" justifyContent=\\"space-between\\">
                    <div style={{...}} className=\\"details__date\\">
                      <div>
                        <b>
                          Created:
                        </b>
                         
                        01/01/70
                      </div>
                      <div>
                        <b>
                          Edited:
                        </b>
                         
                        01/01/70
                      </div>
                    </div>
                  </FlexBox>
                  <FlexBox className=\\"details__tags\\" justifyContent=\\"flex-start\\" flexWrap=\\"wrap\\">
                    <div style={{...}} className=\\"details__tags\\">
                      <Connect(NoteTagBase) tagId=\\"1\\">
                        <NoteTagBase tagId=\\"1\\" tags={{...}} dispatch={[Function]}>
                          <div className=\\"note__tag\\">
                            Tag1
                          </div>
                        </NoteTagBase>
                      </Connect(NoteTagBase)>
                    </div>
                  </FlexBox>
                </div>
              </FlexBox>
            </div>
          </FlexBox>
        </div>
      </ContentContainer>
    </NotesItem>
  </Provider>
</WithFontAwesome>"
`;
